### PR TITLE
validate miner params

### PR DESCRIPTION
**Pull request by Stebalien**
Part of https://github.com/filecoin-project/specs-actors/issues/604

### CODE REVIEW

File: actors/builtin/miner/miner_actor.go [hunk 1]
```diff
@@ -205,6 +206,7 @@ type ChangeMultiaddrsParams struct {
 }
 
 func (a Actor) ChangeMultiaddrs(rt Runtime, params *ChangeMultiaddrsParams) *adt.EmptyValue {
+	// TODO: Max number/size of multiaddrs?
```
Comment (Stebalien) [inline actors/builtin/miner/miner_actor.go:209]: ?

File: actors/builtin/miner/miner_actor.go [hunk 2]
```diff
@@ -187,6 +187,7 @@ type ChangePeerIDParams struct {
 }
 
 func (a Actor) ChangePeerID(rt Runtime, params *ChangePeerIDParams) *adt.EmptyValue {
+	// TODO: Max size of peer ID?
```
Comment (Stebalien) [inline actors/builtin/miner/miner_actor.go:190]: ?

File: actors/builtin/miner/miner_actor.go [hunk 3]
```diff
@@ -416,22 +418,33 @@ func (a Actor) SubmitWindowedPoSt(rt Runtime, params *SubmitWindowedPoStParams)
 // Proposals must be posted on chain via sma.PublishStorageDeals before PreCommitSector.
 // Optimization: PreCommitSector could contain a list of deals that are not published yet.
 func (a Actor) PreCommitSector(rt Runtime, params *SectorPreCommitInfo) *adt.EmptyValue {
```
Comment (Stebalien) [inline actors/builtin/miner/miner_actor.go:420]: Most of these checks were already here, I just put them in field order,and switched to RequireParam. Please check _carefully_.

File: actors/builtin/miner/miner_actor.go [hunk 4]
```diff
@@ -1328,6 +1344,8 @@ func (a Actor) CompactPartitions(rt Runtime, params *CompactPartitionsParams) *a
 
 // Locks up some amount of a the miner's unlocked balance (including any received alongside the invoking message).
 func (a Actor) AddLockedFund(rt Runtime, amountToLock *abi.TokenAmount) *adt.EmptyValue {
+	builtin.RequireParam(rt, amountToLock.Sign() >= 0, "cannot lock up a negative amount of funds")
```
Comment (Stebalien) [inline actors/builtin/miner/miner_actor.go:1347]: We do check this later... and panic. We need to audit uses of Assert.

File: actors/builtin/miner/miner_actor.go (hunk 2 above)
Comment (anorth) [reply actors/builtin/miner/miner_actor.go:190]: If this is still TODO when you merge, please reference the issue in the comment.

Comment (Stebalien) [reply actors/builtin/miner/miner_actor.go:190]: Will do. I was going to try to resolve it now, but I'm not even sure if limits make sense here.

**Pull request merged**